From 781c51f6bc5da29ca879de42efa55128b04d5772 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Fri, 11 Oct 2024 03:34:40 +0000
Subject: [PATCH] shadow: update to ruby 3.x str routines

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 shadow/shadow.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/shadow/shadow.c b/shadow/shadow.c
index 35a77a1..314c280 100644
--- a/shadow/shadow.c
+++ b/shadow/shadow.c
@@ -34,8 +34,8 @@ static VALUE rb_eFileLock;
 static VALUE convert_pw_struct( struct spwd *entry ) 
 {
   return rb_struct_new(rb_sPasswdEntry,
-		      rb_tainted_str_new2(entry->sp_namp),
-		      rb_tainted_str_new2(entry->sp_pwdp),
+		      rb_str_new_cstr(entry->sp_namp),
+		      rb_str_new_cstr(entry->sp_pwdp),
 		      INT2FIX(entry->sp_lstchg),
 		      INT2FIX(entry->sp_min),
 		      INT2FIX(entry->sp_max),
-- 
2.39.2

